tests/test_gpu_detection.py

```
@@ -0,0 +1,38 @@
+        device = _get_device()
+        assert device in ["cpu", "cuda"], f"Device should be 'cpu' or 'cuda', got {device}"
+
+    def test_get_device_fallback_to_cpu(self):
+        """Should fallback to CPU gracefully if CUDA unavailable."""
+        from tldr.semantic import _get_device
+
+        # Even without GPU, should not crash
+        device = _get_device()
+        assert device == "cpu", "Should default to CPU if no CUDA available"
+
+    def test_model_loads_with_detected_device(self, tmp_path):
```

Ruff (0.14.13)

Unused method argument: `tmp_path`

(ARG002)